### PR TITLE
Add requestProgress function to Watch and Watcher interfaces

### PR DESCRIPTION
**Pull request by juliankung-db**
Added a `requestProgress` function to the `Watch` and `Watcher` interfaces to request a WatchResponse be sent to all watchers containing the latest revision number. This can be used to check if a watcher is "caught up" to a particular revision, which is useful when determining if a watch cache is stale by comparing the progress revision to the revision returned in a quorum read.

This was added to the etcd client in https://github.com/etcd-io/etcd/pull/9869 and requested for the jetcd client in https://github.com/etcd-io/jetcd/issues/928

### Commit message below

Added a function that requests a watch stream progress status
be sent in the watch response stream as soon as possible. This is
helpful in situations where an application may want to check the
progress of a watch to determine how up-to-date the watch stream is.

Addresses #928

### CODE REVIEW

File: jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java
Comment (juliankung-db) [inline jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java:105]: Alternatively we can send a single RequestProgress update at this point (we'd need to add a FutureStub property if we go this route) and we can propagate the response to all watchers.

This'll save the etcd server from getting hit with multiple requests if there are multiple watchers

File: jetcd-core/src/main/proto/rpc.proto
Comment (mikehuston) [inline jetcd-core/src/main/proto/rpc.proto:511]: Requests a*

File: jetcd-core/src/main/java/io/etcd/jetcd/Watch.java
Comment (mikehuston) [inline jetcd-core/src/main/java/io/etcd/jetcd/Watch.java:144]: latest revision processed?

File: jetcd-core/src/main/java/io/etcd/jetcd/Watch.java
Comment (mikehuston) [inline jetcd-core/src/main/java/io/etcd/jetcd/Watch.java:215]: Requests the latest revision processed?

File: docs/Watch.md
Comment (mikehuston) [inline docs/Watch.md:51]: Send the latest revision processed by each watcher to all... ?

File: docs/Watch.md
Comment (mikehuston) [inline docs/Watch.md:19]:
```suggestion
The etcd client process watch request with [watch function](#watch-function), process notification with [processEvents function](#processevents-function) , process resume with [resume function](#resume-function), process cancel with [cancelWatch function](#cancelwatch-function) and [requestProgress function](#requestProgress-function).
```

**Review by mikehuston — COMMENTED** [review]
some doc nits, of course defer to maintainers

File: jetcd-core/src/main/proto/rpc.proto
Comment (mikehuston) [reply jetcd-core/src/main/proto/rpc.proto:61]: Not sure if improvement

File: docs/Watch.md
Comment (juliankung-db) [reply docs/Watch.md]: Changed to be similar to what you suggested

File: docs/Watch.md
Comment (juliankung-db) [reply docs/Watch.md]: Changed to be more clear

File: jetcd-core/src/main/proto/rpc.proto
Comment (juliankung-db) [reply jetcd-core/src/main/proto/rpc.proto:61]: I'll leave this for the maintainers to decide. I was following what looked like the proto pattern for RPCs in the file

Comment (lburgazzoli) [reply jetcd-core/src/main/proto/rpc.proto:61]: I'd go with `Progress` so it uses the same pattern

**Review by lburgazzoli — APPROVED** [review]

**Comment by juliankung-db** [comment]
Is there a way to kick off the build job again? It looks like one of the watch tests failed (not the one I created), but the test passes locally. Maybe it's it's just a flaky test?

**Comment by juliankung-db** [comment]
@lburgazzoli would you mind merging when you have the chance? I don't have write access so I can't do it myself

**Pull request merged**
